File: mycollab-ui/src/main/scala/com/mycollab/vaadin/mvp/ViewEvent.scala
package com.mycollab.vaadin.mvp

import com.mycollab.events.ApplicationEvent

/**
  * @author MyCollab Ltd
  * @since 5.1.0
  */
class ViewEvent[E](source: AnyRef, data: E) extends ApplicationEvent(source, data) {

}

object ViewEvent {
  val VIEW_IDENTIFIER: String = "viewevent"
}

File: mycollab-ui/src/main/java/com/mycollab/vaadin/events/ViewEvent.kt
package com.mycollab.vaadin.events

import com.mycollab.events.ApplicationEvent

/**
 * @author MyCollab Ltd
 * @since 6.0.0
 */
class ViewEvent<out B>(source: Any, val data: B) : ApplicationEvent(source) {
    companion object {
        @JvmField val VIEW_IDENTIFIER = "viewEvent"
    }
}